tests/utilities/test_cli.py

```diff
@@ -329,3 +331,93 @@ def test_lightning_cli_config_and_subclass_mode(tmpdir):
     assert config['model'] == cli.config['model']
     assert config['data'] == cli.config['data']
     assert config['trainer'] == cli.config['trainer']
+
+
+def any_model_any_data_cli():
+    LightningCLI(
+        LightningModule,
+        LightningDataModule,
+        subclass_mode_model=True,
+        subclass_mode_data=True,
+    )
+
+
+def test_lightning_cli_help():
+
+    cli_args = ['any.py', '--help']
+    out = StringIO()
+    with mock.patch('sys.argv', cli_args), redirect_stdout(out), pytest.raises(SystemExit):
+        any_model_any_data_cli()
+
+    assert '--print_config' in out.getvalue()
+    assert '--config' in out.getvalue()
+    assert '--seed_everything' in out.getvalue()
+    assert '--model.help' in out.getvalue()
+    assert '--data.help' in out.getvalue()
+
```

Comment:
Related to https://github.com/PyTorchLightning/pytorch-lightning/issues/7192 maybe we could add a check here that *all* trainer parameters are in the help, like below. This would make sure that the CLI always supports all trainer options. Though this could mean that if Trainer is changed then this test fails. @awaelchli @carmocca @Borda what do you think?

```python
for param in inspect.signature(Trainer.__init__).parameters.keys():
    if param != 'self':
        assert f'--trainer.{param}' in out.getvalue()
```

Reply:
Yes, let's do this.

Reply:
Skipping an can be simply changing `if param != 'self'` to `if param not in {'self', 'arg1', ...}`. But this would be a change in the tests if it starts failing.

Reply:
Yes, that's okay